src/testing/core/mocking/__tests__/mockLink.ts

```diff
@@ -0,0 +1,32 @@
+import gql from "graphql-tag";
+import { MockLink } from "../mockLink";
+import { createOperation } from "../../../../link/utils";
+
+describe("mockLink", () => {
+  it("should not require a result or error when delay equals Infinity", async () => {
+    const query = gql`
+      query A {
+        a
+      }
+    `;
+    const mockLink = new MockLink([
+      {
+        request: {
+          query,
+        },
+        delay: Infinity,
+      },
+    ]);
+
+    const observable = mockLink.request(createOperation(null, { query }));
```

Comment:
```suggestion
    const observable = execute(mockLink, { query });
```

Can you use the `execute` function from [`link/core/execute`](https://github.com/apollographql/apollo-client/blob/main/src/link/core/execute.ts) here instead? It does some of this setup for you and ensures the link is called like it would be in production 🙂.

Normally I wouldn't care too much, but since this is a new file, patterns from other tests tend to multiply, and I'd like to ensure we use `execute` where possible.